common/k8s/src/middleware/metadata.rs

```diff
@@ -180,12 +180,19 @@ impl Middleware for K8sMetadata {
         });
     }
 
-    fn process(&self, mut line: LineBuilder) -> Status {
-        if let Some(ref file_name) = line.file {
+    fn process<'a>(&self, line: &'a mut dyn LineMetaMut) -> Status<&'a mut dyn LineMetaMut> {
```

Comment:
We calling per each `LazyLineSerializer` but the annotations should be the same per each `TailedFile`.

Reply:
Maybe it makes sense to inject this into the Tailers somehow? rather than applying it to the lines after the fact?

Reply:
or when creating a `LazyLineSerializer` we could try to copy the previous values and check here if the annotations/labels are already filed in.

Reply:
We need a way to update them over time, I suppose we could change TailedFile for the LazyLineSerializer to subscribe directly to the k8s metadata and maintain a template internally which it can update and clone? Then we'd only be changing the template when we need to and clones should be quick.